### PR TITLE
[GraphBolt] add available builtin datasets into docstring

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/impl/ondisk_dataset.py [hunk 1]
```diff
@@ -482,6 +482,17 @@ class BuiltinDataset(OnDiskDataset):
     This class is used to help download datasets from DGL S3 storage and load
     them as ``OnDiskDataset``.
 
+    Available builtin datasets include:
+    **ogbn-mag**
+        The ogbn-mag dataset is a heterogeneous network composed of a subset of
+        the Microsoft Academic Graph (MAG). See more details in
+        `ogbn-mag <https://ogb.stanford.edu/docs/nodeprop/#ogbn-mag>`_.
+    **ogbl-citation2**
```
Comment (frozenbugs) [inline python/dgl/graphbolt/impl/ondisk_dataset.py:490]: Worth to callout the pre-process on top of the data if we did.

File: python/dgl/graphbolt/impl/ondisk_dataset.py [hunk 2]
```diff
@@ -491,10 +502,15 @@ class BuiltinDataset(OnDiskDataset):
...
 
     _base_url = "https://data.dgl.ai/dataset/graphbolt/"
+    _datasets = ["ogbn-mag", "ogbl-citation2"]
 
     def __init__(self, name: str, root: str = "datasets") -> OnDiskDataset:
         dataset_dir = os.path.join(root, name)
         if not os.path.exists(dataset_dir):
+            if name not in self._datasets:
+                raise RuntimeError(
+                    f"Dataset {name} is not available. Available datasets are {self._datasets}"
```
Comment (frozenbugs) [inline python/dgl/graphbolt/impl/ondisk_dataset.py:512]: nit: over 80 chars.

**Review by frozenbugs — APPROVED** [review]

File: python/dgl/graphbolt/impl/ondisk_dataset.py [hunk 3]
```diff
@@ -491,10 +514,16 @@ class BuiltinDataset(OnDiskDataset):
     """
 
     _base_url = "https://data.dgl.ai/dataset/graphbolt/"
+    _datasets = ["ogbn-mag", "ogbl-citation2"]
```
Comment (mfbalin) [inline python/dgl/graphbolt/impl/ondisk_dataset.py:517]: ogbn-products is missing

Comment (Rhett-Ying) [reply python/dgl/graphbolt/impl/ondisk_dataset.py:517]: thanks for pointing this out

**Pull request merged**